### ISSUE TITLE
vim debugger add inline value

### ISSUE THREAD

**Issue by vertices-cpp** [vim/vim#10076]
Can you increase termdebug inline values
![v2-19d9f2c199b7a842bacbd5538de0e793_b](https://user-images.githubusercontent.com/90932460/161453720-a41431c3-a773-45c4-9a40-689e25148ba9.jpg)

**Comment by chrisbra** [comment]
Hi, please use this opportunity to describe in a bit more detail what exactly you want. For me it's not clear.

**Comment by QQ1598058687** [comment]
debugging of vim,
packadd termdebug
Termdebug 
When debugging c files, isn't there a watch variable? Can you display the debugged variables to the current right side of the source code?

**Comment by brammool** [comment]
Not yet.  I thought someone was working on showing watched variables (or expression), but I haven't kept track.
We can see values by hovering over an ID, using that mechanism to show certain values should be possible.
Requires some way to add and remove these.

**Comment by QQ1598058687** [comment]
https://imgur.com/gallery/mcd5H10
![Uploading 109373505-0cb23380-7864-11eb-8bf2-a991873b6d02.png…]()
I just realized the mouse touch screen in the previous paragraph, but I don't understand the text control
Need to display variable value inline

**Comment by chrisbra** [comment]
Please use vi.stackexchange.com or the vim-use mailinglist for questions. We cannot provide individual help here.

**Comment by QQ1598058687** [comment]
Yes, when most people choose an editor as a tool, due consideration should be given to certain functions in the same field

**Comment by habamax** [comment]
> It should be possible to use a popup window to show variable values. But it requires a way to decide what values to show (perhaps by clicking on it?).

Ale uses popup windows to show inline error messages. Not sure if it is simple enough to extend the usage to show many popup windows with some calculated values though.

![image](https://user-images.githubusercontent.com/234774/161921410-936f574d-73da-40d6-9cdb-52280a06b09b.png)

**Comment by GitMensch** [comment]
Note: this feature request is about showing _all_ expressions the debugger can handle "inline" (that's an option with the vscode debugger from the screenshot).

I'm not convinced that this is a feature that wouldn't slow down everything a lot; but it _may_ be useful for some places. I _think_ this is most reasonable when an LSP already tells the editor "that's an expression that is interesting".

An alternative "inline display values" I've seen was to only show the expressions in the current line (and/or +/- N context lines), but then again there's the need for something to say "these are the expressions seen in the line".

The current option of showing values on hover works quite fine (at least if you don't use Putty #3208).

The feature request to _watch_ / monitor expressions is tracked as #9065. I had a rough proof-of-concept using a list in a separate window (similar to the assembler); adding to it with ":Monitor <expression>", that may be triggered from the editor like ":Evaluate"; having an inline display for those would be very useful - but not all expressions that should be monitored are found in the text as-is, so maybe "both" inline and/or in a window would be reasonable?

Note: for my cases I've found the most important thing to have the monitor list to store the expressions by stack frame (function name) to not show these when they are currently out of scope and to show them again when re-entering the function.

In any case: someone helping with the implementation or working on it (possibly "only the expressions from the code as popup window with a mapping key to add/remove it") would be marvelous. I can give input and test...

**Comment by lacygoill** [comment]
> [...] someone helping with the implementation [...] would be marvelous. I can give input and test...

I can't implement any feature, because I don't use the plugin, but if you have specific goals/inputs/tests, I *might* help.

**Comment by GitMensch** [comment]
What we had since a while is a manual request, pressing K on a word (or a marked expression).
@BigPeet added the option to use a popup window at that place instead of displaying the value (so something similar what the ale screenshot showed) last year, but that only works if it is clear where to use it.

*If* vim _knows_ what the actual expressions are (for example a line that has `c[x] = d[y]` would have the relevant expressions `x`, `y`, `c[x]` and `d[y]` to show, then the debugger could be queried to resolve those (maybe with a maximal length like 10) and be shown in either a popup window or "real inline" (not sure how that would work, I guess a popup window with bg matching the normal bg and fg being the mix between bg and fg).
If vim doesn't know about the expressions it may "brute force" (hammering) the debugger with requests of any single word (if it knows about comments then leaving those out), but for performance reasons this is likely only useful for the current active line.
